src/Opaleye/Internal/PGTypesExternal.hs

```diff
@@ -182,6 +182,15 @@ pgRange pgEl start end =
         oneEl R.NegInfinity   = HPQ.NegInfinity
         oneEl R.PosInfinity   = HPQ.PosInfinity
 
+-- Full Text Search
+
+pgTSVector :: String -> Field PGTSVector
+pgTSVector = IPT.castToType "tsvector" . HSD.quote
+
+pgTSQuery :: String -> Field PGTSQuery
+pgTSQuery = IPT.castToType "tsquery" . HSD.quote
```

Comment:
Shouldn't these take `Field PGText` as argument, rather than `String`? Otherwise you can't do text search on anything that's actually available in your database.

Reply:
Yes, that's a good idea, thanks.

Reply:
Ah, ok, but a question: how should I proceed with destructuring `PGText`? I found `unColumn` but it's not enough. When constructing the `String -> Field PGTSVector` type, I used as example the `pgJSON` etc functions in the same file.

Reply:
I guess `FunExpr`, for example https://github.com/tomjaguarpaw/haskell-opaleye/blob/912641cddd98bbdd315f5614b4098dc2460064ee/src/Opaleye/Operators.hs#L291-L292